docs/logging.md

```diff
@@ -0,0 +1,204 @@
+  looking, but its syntax is more error-prone (no logging in case you forget
+  to call `Msg("...")` at the end)
+- Go's [Slog](https://pkg.go.dev/log/slog) is new, slower (according to Zap
+  benchmarks), and doesn't provide useful benefits
+
+## Default Level
+
+The default minimum logging level is [Info](https://pkg.go.dev/go.uber.org/zap/zapcore#InfoLevel).
+To change this, set the following AutoKitteh config value to the lower-case or
+all-caps name of the level you want. For example:
+
+- CLI: `ak config set logger.level debug`
+
+- Config YAML file:
+
+  ```yaml
+  logger:
+    level: debug
+  ```
+
+- Environment variable: `AK_LOGGER__LEVEL=DEBUG`
+
+## Logger Naming
+
+- Local variables: `l` is preferable when its meaning is clear, otherwise
+  `logger` is acceptable too
+
+  - `z` is discouraged - we use Zap, but we don’t want the code to be tied to
+    that assumption
+  - Other combinations of `l` and `logger` are possible for code readability,
+    but not recommended in general
+
+- Struct fields: `logger` or `Logger`
+
+## Logger Initialization
+
+> [!NOTE]
+> TODO: Refactor `extrazap`
+>
+> TODO: Godoc link below
+
+Pass a logger reference (`*zap.Logger`) as a function parameter, or use
+[extrazap](/TODO) to initialize a logger instance:
+
+```go
+func one(l *zap.Logger, ...) {
+    l.Info("...")
+
+    ctx := extrazap.AttachLoggerToContext(context.Background(), l)
+    two(ctx, ...)
+}
+
+func two(ctx context.Context, ...) {
+    l := extrazap.ExtractLoggerFromContext(ctx)
+    l.Info("...")
+}
+```
+
+You can use `extrazap` even if your function doesn't have a context with an
+embedded logger:
+
+```go
+func three() {
+    l := extrazap.ExtractLoggerFromContext(context.Background())
+    l.Info("...")
+}
+```
+
+In other words, you **should** attach a logger to a context before passing it
+to a function that uses `extrazap`, but it will work even if you don't.
+
+Because of that, there's no need to call [zap.L()](https://pkg.go.dev/go.uber.org/zap#L)
+directly.
+
+Anyway, do not pass both a context and a logger as function parameters, this
+is wasteful.
+
+## Log Messages
+
+1. Should be short
+   - Avoid complex or lengthy sentences
+   - Prefer adding details as fields, not in the message string
+2. Should be informative - if the message is too generic or vague, add vital
+   details to it (in addition to adding them as fields)
+3. Should begin with a capital letter
+   - This is unlike Go error strings, but for the
+     [same reason](https://go.dev/wiki/CodeReviewComments#error-strings)
+4. Shouldn’t end with punctuation
+   - This is like Go error strings, but only to support (1) above
+
+## Adding Log Fields to the Logger
+
+As noted above, we always use loggers by reference (`*zap.Logger`), not by
+value (`zap.Logger`).
+
+Because of that, when you call [Logger.With()](https://pkg.go.dev/go.uber.org/zap#Logger.With)
+in a nested code block, redeclare the logger with the `:=` operator:
```

Comment:
Show a full example. IMO this doesn't happen a lot, most cases the same logger will be use throughout the function.